Add disable-builtin-k8s-secret-store annotation option (#2548)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/secrets/secrets-overview.md

1. Retrieving secrets using the Dapr secrets API in the application code.
1. Optionally, referencing secrets in Dapr component files.

{{% alert title="Note" color="primary" %}}

> By default, Dapr currently enables a built-in Kubernetes secret store in Kubernetes mode (deployed via the Helm defaults or `dapr init -k`). If you are using another secret store, you can disable the Dapr Kubernetes secret store using the `disable-builtin-k8s-secret-store` setting, when set to `true`. The default is `false`.

Reviewer: ```suggestion
By default, Dapr enables a built-in [Kubernetes secret store in Kubernetes mode]({{<ref "kubernetes-secret-store.md">}}) deployed either via the Helm defaults or `dapr init -k`. If you are using another secret store, you can disable (not configure) the Dapr Kubernetes secret store using the `disable-builtin-k8s-secret-store` setting, when set to `true` through the Helm settings. The default is `false`.
```

1. Retrieving secrets using the Dapr secrets API in the application code.
1. Optionally, referencing secrets in Dapr component files.

> Note: Dapr currently, by default, enables a built-in kubernetes secret store in kubernetes mode.

Reviewer 2: ```suggestion
{{% alert title="Note" color="primary" %}}
By default, Dapr currently enables a built-in Kubernetes secret store in Kubernetes mode. If you don't want to use any secrets API or would rather use a different secret store, you can disable the default Kubernetes secret store using the new `disable-builtin-k8s-secret-store` setting, which can be set to `true`. The default setting is `false`, to not disturb the ongoing default behavior of Dapr.
{{% /alert %}}
```

Encapsulating the whole note in shortcode

Reviewer: I would suggest a slight variant like this.

```suggestion
{{% alert title="Note" color="primary" %}}
By default, Dapr currently enables a built-in Kubernetes secret store in Kubernetes mode (deployed via the Helm defaults or `dapr init -k`). If you are using another secret store, you can disable the Dapr Kubernetes secret store using the `disable-builtin-k8s-secret-store` setting, when set to `true`. The default is `false`.
{{% /alert %}}
```

Author: Thanks @hhunter-ms  and @msfussell. Incorporated suggestions.
